### PR TITLE
You can select country and region

### PR DESCRIPTION
**Pull request by OscarSanner**
## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [x] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] I have followed the [perfect PR recommendations][perfect-pr]
- [x] The code has been formatted using Black (`black --fast homeassistant tests`)
- [x] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [x] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [x] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [x] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [x] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.
- [x] Untested files have been added to `.coveragerc`.

To help with the load of incoming pull requests:

- [x] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

[dev-checklist]: https://developers.home-assistant.io/docs/development_checklist/
[manifest-docs]: https://developers.home-assistant.io/docs/creating_integration_manifest/
[quality-scale]: https://developers.home-assistant.io/docs/integration_quality_scale_index/
[docs-repository]: https://github.com/home-assistant/home-assistant.io
[perfect-pr]: https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr

### CODE REVIEW

File: homeassistant/components/openAQ/config_flow.py [hunk 1]
```diff
@@ -25,39 +21,99 @@ class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
...
-        res = get_device(
-            location_id=user_input[LOCATION_ID], api_key=user_input[API_KEY_ID]
-        ).sensors
+        name = get_device(location_id=self.location_id, api_key=self.api_key).locality
 
-        name = get_device(
-            location_id=user_input[LOCATION_ID], api_key=user_input[API_KEY_ID]
-        ).locality
+        if name is None:
+            name = {i for i in locations_dict if locations_dict[i] == self.location_id}
```
Comment (Ronserro) [inline homeassistant/components/openAQ/config_flow.py:85]: what is "i" in this case?

File: homeassistant/components/openAQ/const.py [hunk 2]
```diff
@@ -7,6 +7,9 @@
 
 API_KEY_ID = "api_id"
 LOCATION_ID = "location_id"
+THIRD_FIELD = "test"
```
Comment (Ronserro) [inline homeassistant/components/openAQ/const.py]: What is this?

File: homeassistant/components/openAQ/config_flow.py [hunk 3]
```diff
@@ -25,39 +21,99 @@ class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
...
+            self.location_id = locations_dict[user_input[LOCATION]]
+
+        self._data = {API_KEY_ID: self.api_key, LOCATION_ID: self.location_id}
+
+        res = get_device(location_id=self.location_id, api_key=self.api_key).sensors
 
-        res = get_device(
-            location_id=user_input[LOCATION_ID], api_key=user_input[API_KEY_ID]
-        ).sensors
+        name = get_device(location_id=self.location_id, api_key=self.api_key).locality
```
Comment (Ronserro) [inline homeassistant/components/openAQ/config_flow.py]: can you make only one get_device call as it sends an api request every time

File: homeassistant/components/openAQ/config_flow.py (hunk 1 above)
Comment (OscarSanner) [reply homeassistant/components/openAQ/config_flow.py:85]: it's an index variable for iterating over all dictionary entries

Comment (Ronserro) [reply homeassistant/components/openAQ/config_flow.py:85]: why not use "if self.location_id in locations_dict" or is multiple values expected?

File: homeassistant/components/openAQ/const.py (hunk 2 above)
Comment (OscarSanner) [reply homeassistant/components/openAQ/const.py]: its gone now

File: homeassistant/components/openAQ/config_flow.py (hunk 3 above)
Comment (OscarSanner) [reply homeassistant/components/openAQ/config_flow.py]: yeah

Comment (OscarSanner) [reply homeassistant/components/openAQ/config_flow.py]: fixed

**Pull request merged**